test/fakeApiGatewayEvent.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  APIGatewayProxyEvent,
+  APIGatewayEventRequestContext,
+  APIGatewayEventDefaultAuthorizerContext,
+  APIGatewayEventIdentity,
+} from 'aws-lambda';
+
+export function createEvent(body:any): APIGatewayProxyEvent {
+  const auth: APIGatewayEventDefaultAuthorizerContext = {};
+  const identity: APIGatewayEventIdentity = {
+    accessKey: '',
+    accountId: '',
+    apiKey: '',
+    apiKeyId: '',
+    caller: '',
+    cognitoAuthenticationProvider: '',
+    cognitoAuthenticationType: '',
+    cognitoIdentityId: '',
+    cognitoIdentityPoolId: '',
+    principalOrgId: '',
+    sourceIp: '',
+    user: '',
+    userAgent: '',
+    userArn: '',
+  };
+
+  const request: APIGatewayEventRequestContext = {
+    accountId: '',
+    apiId: '',
+    authorizer: auth,
+    connectedAt: 0,
+    connectionId: '',
+    domainName: '',
+    domainPrefix: '',
+    eventType: '',
+    extendedRequestId: '',
+    protocol: '',
+    httpMethod: '',
+    identity: identity,
+    messageDirection: '',
+    messageId: '',
+    path: '',
+    stage: '',
+    requestId: '',
+    requestTime: '',
+    requestTimeEpoch: 0,
+    resourceId: '',
+    resourcePath: '',
+    routeKey: '',
+  };
+
+  const bodyText = JSON.stringify(body);
+
+  const evt: APIGatewayProxyEvent = {
+    body: bodyText,
+    headers: {},
+    multiValueHeaders: {},
+    httpMethod: 'POST',
+    isBase64Encoded: false,
+    path: 'ProxyMgiStatus',
+    requestContext: request,
```

Comment:
use type casting to have minimal mocking
```suggestion
    requestContext: {} as unknown as APIGatewayEventRequestContext,
```

Reply:
done, great, great tip, thanks